src/ddscxx/src/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.cpp

```diff
@@ -160,6 +168,7 @@ DataReaderQosDelegate::ddsc_qos(const dds_qos_t* qos)
     reliability_ .delegate().set_iso_policy(qos);
     resources_   .delegate().set_iso_policy(qos);
     user_data_   .delegate().set_iso_policy(qos);
+    datarepresentation_.delegate().set_iso_policy(qos);
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).

Reply:
Will do.